prov/util/src/util_attr.c

```diff
@@ -162,8 +162,19 @@ static int ofi_info_to_core(uint32_t version, const struct fi_provider *prov,
 	const char *core_name;
 	size_t len;
 
-	if (!(*core_hints = fi_allocinfo()))
-		return -FI_ENOMEM;
+	/* If we are here through fi_fabric we must allocate only fabric_attr */
+	if (util_info && util_info->fabric_attr && util_info->fabric_attr->name &&
+	    !util_info->domain_attr) {
```

Comment:
Agree with you. Perhaps we should use fi_dupinfo which doesn't alloc other attributes if it's not present in the util_hints.